tools/github-actions/download-build-output/action.yml

```diff
@@ -5,9 +5,33 @@ runs:
   using: 'composite'
   steps:
     - uses: actions/download-artifact@v3
+      if: github.event_name != 'workflow_run'
       with:
         name: dist
         path: '.'
+
+    - name: 'Download artifact'
+      uses: actions/github-script@v6
+      if: github.event_name == 'workflow_run'
+      with:
+        script: |
+          let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+             owner: context.repo.owner,
+             repo: context.repo.repo,
+             run_id: context.payload.workflow_run.id,
+          });
+          let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+            return artifact.name == "dist"
+          })[0];
+          let download = await github.rest.actions.downloadArtifact({
```

Comment:
as it is actually an exposed Github action, maybe you should handle le case of `matchArtifact == undefined` (and maybe provide an input for the `artifact name`)

Reply:
done